backend/controller/dal/encryption.go

```diff
@@ -58,49 +58,31 @@ func (d *DAL) decryptJSON(subKey encryption.SubKey, encrypted []byte, v any) err
 	return nil
 }
 
-// setupEncryptor sets up the encryptor for the DAL.
-// It will either create a key or load the existing one.
-// If the KMS URL is not set, it will use a NoOpEncryptor which does not encrypt anything.
-func (d *DAL) setupEncryptor(ctx context.Context) (err error) {
+func (d *DAL) EnsureKey(ctx context.Context, generateKey func() ([]byte, error)) (encryptedKey []byte, err error) {
 	logger := log.FromContext(ctx)
 	tx, err := d.Begin(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to begin transaction: %w", err)
+		return nil, fmt.Errorf("failed to begin transaction: %w", err)
 	}
 	defer tx.CommitOrRollback(ctx, &err)
 
-	url, ok := d.kmsURL.Get()
-	if !ok {
-		logger.Infof("KMS URL not set, encryption not enabled")
-		d.encryptor = encryption.NewNoOpEncryptor()
-		return nil
-	}
+	encryptedKey, err = tx.db.GetOnlyEncryptionKey(ctx)
+	if err != nil && dal.IsNotFound(err) {
+		logger.Infof("No encryption key found, generating a new one")
```

Comment:
Let's make this debug. In general we're putting most logs at debug, so they don't show up for the user during development mode.